### PR TITLE
snapshotter: recompute hashes if file has changed

### PR DESCRIPTION
**Pull request by mhoffm-aiven**

### CODE REVIEW

File: astacus/node/snapshotter.py [hunk 1]
```diff
@@ -101,27 +101,32 @@ def _snapshotfile_from_path(self, relative_path):
...
-            yield snapshotfile
+            if old_snapshotfile and self._files_are_the_same(old_snapshotfile, new_snapshotfile):
+                new_snapshotfile.hexdigest = old_snapshotfile.hexdigest
+                new_snapshotfile.content_b64 = old_snapshotfile.content_b64
+                same += 1
+                if increase_worth_reporting(same):
+                    logger.debug("#%d. same - %r in %s is same", same, old_snapshotfile, relative_path)
+                continue
+            else:
+                new_snapshotfile = self._compute_and_set_hash(snapshotfile=new_snapshotfile)
```
Comment (fingon) [inline astacus/node/snapshotter.py:127]: This is unnecessary. The returned snapshotfiles are changed and hashed anyway later so you're refactoring more than you need to.

**Review by fingon — DISMISSED** [review]

File: astacus/node/snapshotter.py (hunk 1 above)
Comment (mhoffm-aiven) [reply astacus/node/snapshotter.py]: :heavy_check_mark:  thanks i removed that part and tests are still passing

File: astacus/node/snapshotter.py [hunk 2]
```diff
@@ -101,27 +101,30 @@ def _snapshotfile_from_path(self, relative_path):
         st = src_path.stat()
         return SnapshotFile(relative_path=relative_path, mtime_ns=st.st_mtime_ns, file_size=st.st_size)
 
+    def _files_are_the_same(self, old: SnapshotFile, new: SnapshotFile) -> bool:
```
Comment (fingon) [inline astacus/node/snapshotter.py:104]: nit: This should be probably something in SnapshotFile itself.

File: tests/unit/node/test_node_snapshot.py [hunk 3]
```diff
@@ -150,3 +150,27 @@ def _fun(self):
     progress = response.json()["progress"]
     assert progress["failed"]
     assert progress["final"]
+
+
+@pytest.mark.timeout(2)
+def test_snapshot_shrinked_file_too_small(snapshotter):
```
Comment (fingon) [inline tests/unit/node/test_node_snapshot.py:156]: nit: isn't it shrunk? :p

File: tests/unit/node/test_node_snapshot.py [hunk 4]
```diff
@@ -150,3 +150,27 @@ def _fun(self):
...
+    PATH = snapshotter.src / "shrinky1"
```
Comment (fingon) [inline tests/unit/node/test_node_snapshot.py:157]: Why is this uppercase? They are usually globals.

**Review by fingon — CHANGES_REQUESTED** [review]

File: tests/unit/node/test_node_snapshot.py (hunk 3 above)
Comment (mhoffm-aiven) [reply tests/unit/node/test_node_snapshot.py:156]: i think you are right!

File: astacus/node/snapshotter.py (hunk 2 above)
Comment (mhoffm-aiven) [reply astacus/node/snapshotter.py]: :heavy_check_mark: done

File: tests/unit/node/test_node_snapshot.py (hunk 3 above)
Comment (mhoffm-aiven) [reply tests/unit/node/test_node_snapshot.py]: done

File: tests/unit/node/test_node_snapshot.py (hunk 4 above)
Comment (mhoffm-aiven) [reply tests/unit/node/test_node_snapshot.py]: parameterized the tests

**Review by fingon — APPROVED** [review]

**Pull request merged**